doc/dev.rst

```diff
@@ -0,0 +1,54 @@
+Development Guidelines
+----------------------
+
+The idea behind this repository is to keep track of the CHIME pipeline
+development, such that the union of the input data and this repository always
+gives the same output. This requires that we keep track of not only the code and
+scripts in this repository, but also any dependencies (discussed below).
+
+As far as possible the pipeline code should be using Kiyo's pipeline task module `caput.pipeline`  ([doc](http://bao.phas.ubc.ca/codedoc/caput/)).
+
+Structure
+^^^^^^^^^
+
+Tasks should go into `ch_pipeline`.
+
+Branches
+^^^^^^^^
+
+Development should be done in `passX` branches, or in feature branches that are
+merged back in.
+
+Dependencies
+^^^^^^^^^^^^
+
+Dependencies should be installable python packages. This means that they must
+have a ``setup.py`` script in their root directory, and should be installable
+using `python setup.py install`. They are kept track of in a ``requirements.txt``
+file. This file can contain references to exact versions of dependencies, using
+both version tags, and commit hashes. An example ``requirements.txt`` file is
+given below::
+
+    -e git+https://github.com/radiocosmology/caput@ee1c55ea4cf8cb7857af2ef3adcb2439d876768d#egg=caput-master
```

Comment:
I think we want to discourage installing in `editable mode`! Because then pip interprets the version as the commit hash.

I think we also want to discourage `#egg`, my understanding is that it is deprecated.